pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -166,25 +165,28 @@ func (m *Multicluster) close() (err error) {
 // AddMemberCluster is passed to the secret controller as a callback to be called
 // when a remote cluster is added.  This function needs to set up all the handlers
 // to watch for resources being added, deleted or changed on remote clusters.
-func (m *Multicluster) AddMemberCluster(client kubelib.Client, clusterID string) error {
+func (m *Multicluster) AddMemberCluster(clusterID string, rc *secretcontroller.Cluster) error {
```

Comment:
It would also require splitting out `SyncTimeout`, if that's fine? I think it's a bit more readable to store all the info related to the remote cluster in the struct.

Reply:
maybe a follow up, just seems a little weird